ably.ts.d

```diff
@@ -0,0 +1,400 @@
+  get: (name: String, channelOptions?: ChannelOptions) => RealtimeChannel;
+
+  release: (name: String) => void;
+}
+
+export declare class Message {
+  constructor();
+
+  fromEncoded: (JsonObject: String, channelOptions: ChannelOptions) => Message;
+
+  fromEncodedArray: (JsonArray: String, channelOptions: ChannelOptions) => Array<Message>;
+
+  clientId: String;
+  connectionId: String;
+  data: any;
+  encoding: String;
+  extras: String;
+  id: String;
+  name: String;
+  timestamp: number; // Time
+}
+
+export declare class PresenceMessage {
+  fromEncoded: (JsonObject: any, channelOptions?: ChannelOptions) => PresenceMessage;
+
+  fromEncodedArray: (JsonArray: Array<any>, channelOptions?: ChannelOptions) => Array<PresenceMessage>;
+
+  action: PresenceAction;
+  clientId: String;
+  connectionId: String;
+  data: any;
+  encoding: String;
+  id: String;
+  timestamp: number; // Time
+}
+
+export declare class Rest {
+  constructor(options: String | ClientOptions);
+  auth: Auth;
+  channels: Channels;
+  request: (method: string, path: string, parmas?: any, body?: Array<any> | any, headers?: any, callback?: (Error: ErrorInfo, results: HttpPaginatedResponse) => void) => void;
+  stats: (ParamsOrCallback?: any, callback?: PaginatedResultCallback) => void;
+  time: (paramsOrCallback?: any, callback?: (Error: ErrorInfo, time: number ) => void) => void; // time
+}
+
+export declare class Realtime {
+  constructor(options: String | ClientOptions);
+
+  auth: Auth;
+  channels: Channels;
+  clientId: String;
+  connection: Connection;
+
+  request: (method: string, path: string, parmas?: any, body?: Array<any> | any, headers?: any, callback?: (Error: ErrorInfo, results: HttpPaginatedResponse) => void) => void;
+
+  stats: (ParamsOrCallback?: any, callback?: PaginatedResultCallback) => void;
+
+  close: () => void;
+
+  connect: () => void;
+
+  time: (paramsOrCallback?: any, callback?: (Error: ErrorInfo, time: number ) => void) => void; // time
+
+}
+
+export declare class Connection extends EventEmitter {
+  errorReason: ErrorInfo;
+  id: String;
+  key: String;
+  recoveryKey: String;
+  serial: number;
+  state: ConnectionState;
+
+  close: () => void;
+
+  connect: () => void;
+
+  ping: (callback?: (Error: ErrorInfo, ResponseTime: number ) => void ) => void;
+}
+
+export declare class Stats {
+  all: StatsMessageTypes;
+  apiRequests: StatsRequestCount;
+  channels: StatsResourceCount;
+  connections: StatsConnectionTypes;
+  inbound: StatsMessageTraffic;
+  intervalId: String;
+  outbound: StatsMessageTraffic;
+  persisted: StatsMessageTypes;
+  tokenRequests: StatsRequestCount;
+}
+
+
+export declare class PaginatedResult {
+  items: Array<any>;
+
+  first: (PaginatedResultCallback) => void;
+
+  next: (PaginatedResultCallback) => void;
+
+  current: (PaginatedResultCallback) => void;
+
+  hasNext: () => boolean;
+
+  isLast: () => boolean;
+
+}
+
+export declare class HttpPaginatedResponse extends PaginatedResult {
```

Comment:
If the type is included using generics, then the `items` won't need to be overwritten, but more importantly, according to this, `next()` will now return a `PaginatedResult` instead of a `HttpPaginatedResponse` which is wrong.